code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/EdgePersonalizationResponseHandler.java

```diff
@@ -184,104 +202,217 @@ void handleEdgePersonalizationNotification(final Event edgeResponseEvent) {
+                inMemoryFeeds.remove(surface);
+            }
+        }
+    }
+
+    private Map<String, Feed> processFeedConsequences(final List<RuleConsequence> feedConsequences) {
+        if (feedConsequences.isEmpty()) {
+            Log.debug(LOG_TAG, SELF_TAG, "Will not process feed consequences as none were found in the rules engine.");
+            return null;
+        }
+
+        final Map<String, Feed> processedFeeds = new HashMap<>();
+        for (final RuleConsequence consequence : feedConsequences) {
+            final Map details = consequence.getDetail();
+            final FeedItem feedItem = createFeedItemFromConsequenceDetail(details);
+            final String appSurface = getAppSurface();
+            Feed feed = processedFeeds.get(appSurface);
+            final List<FeedItem> feedItems = feed == null ? new ArrayList<>() : feed.getItems();
+            feedItems.add(feedItem);
+            feed = new Feed(appSurface, feedItems);
+            processedFeeds.put(appSurface, feed);
+        }
+
+        return processedFeeds;
     }
 
     /**
-     * Attempts to load in-app message rules contained in the provided {@code List<PropositionPayload>}. Any valid rule {@link JSONObject}s
-     * found will be registered with the {@link LaunchRulesEngine}.
+     * Attempts to parse any in-app message or message feed rules contained in the provided {@code List<Proposition>}. Any valid rule {@code JSONObject}s
+     * found will be returned in a {code List<LaunchRule>}.
      *
-     * @param propositions       A {@link List<PropositionPayload>} containing in-app message definitions
-     * @param clearExistingRules {@code boolean} if true the existing cached propositions are cleared and new message rules are replaced in the {@code LaunchRulesEngine}
-     * @param persistChanges     {@code boolean} if true the passed in {@code List<PropositionPayload>} are added to the cache
-     * @param expectedScope      {@code String} containing the app surface present in the {@code List<PropositionPayload>}
+     * @param propositions     A {@link List<Proposition>} propositions to be processed
+     * @param expectedSurfaces A {@link List<String>} containing the expected app surfaces
+     * @param clearExisting    {@code boolean} if true the existing cached propositions are cleared and new message rules are replaced in the {@code LaunchRulesEngine}
+     * @param persistChanges   {@code boolean} if true the passed in {@code List<Proposition>} are added to the cache
+     * @return {@link List<LaunchRule>} containing {@link LaunchRule}s parsed from the passed in propositions
      */
-    private void processPropositions(final List<PropositionPayload> propositions, final boolean clearExistingRules, final boolean persistChanges, final String expectedScope) {
+    private List<LaunchRule> processPropositions(final List<Proposition> propositions, final List<String> expectedSurfaces, final boolean clearExisting, final boolean persistChanges) {
```

Comment:
The propositions parsing logic here doesn't account for code-based propositions, please refer to the iOS implementation.